src/solc_0.8/common/BaseWithStorage/SuperOperatorAccessControl.sol

```diff
@@ -0,0 +1,35 @@
+//SPDX-License-Identifier: MIT
+// solhint-disable-next-line compiler-version
+pragma solidity 0.8.2;
+
+import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+
+contract SuperOperatorAccessControl is AccessControlUpgradeable {
+    bytes32 public constant SUPER_OPERATOR_ROLE = keccak256("SUPER_OPERATOR");
+
+    event SuperOperator(address superOperator, bool enabled);
+
+    modifier onlyAdmin() {
+        require(hasRole(DEFAULT_ADMIN_ROLE, _msgSender()), "ADMIN_ONLY");
+        _;
+    }
+
+    /// @notice Enable or disable the ability of `superOperator` to transfer tokens of all (superOperator rights).
+    /// @param superOperator address that will be given/removed superOperator right.
+    /// @param enabled set whether the superOperator is enabled or disabled.
+    function setSuperOperator(address superOperator, bool enabled) external {
+        if (enabled) {
+            grantRole(SUPER_OPERATOR_ROLE, superOperator);
+        } else {
+            revokeRole(SUPER_OPERATOR_ROLE, superOperator);
+        }
+        emit SuperOperator(superOperator, enabled);
+    }
+
+    /// @notice check whether address `who` is given superOperator rights.
+    /// @param who The address to query.
+    /// @return whether the address has superOperator rights.
+    function isSuperOperator(address who) public view returns (bool) {
+        return hasRole(SUPER_OPERATOR_ROLE, who);
+    }
+}
```

Comment:
plus besoin de ce fichier

Reply:
You want to get all the role code in ERC20Base?